crytic_compile/platform/foundry.py

```diff
@@ -118,68 +118,62 @@ def is_supported(target: str, **kwargs: str) -> bool:
-            default_profile = foundry_toml["profile"]["default"]
-
-            def lookup_by_keys(keys: List[str], dictionary: Dict[str, T]) -> Optional[T]:
-                for key in keys:
-                    if key in dictionary:
-                        return dictionary[key]
-                return None
-
-            # Foundry supports snake and kebab case.
-            result.solc_version = lookup_by_keys(
-                ["solc", "solc_version", "solc-version"], default_profile
-            )
-            via_ir = lookup_by_keys(["via_ir", "via-ir"], default_profile)
-            if via_ir:
-                result.via_ir = via_ir
-            result.allow_paths = lookup_by_keys(["allow_paths", "allow-paths"], default_profile)
-
-            if "offline" in default_profile:
-                result.offline = default_profile["offline"]
-            if "optimizer" in default_profile:
-                result.optimizer = default_profile["optimizer"]
-            else:
-                # Default to true
-                result.optimizer = True
-            optimizer_runs = lookup_by_keys(["optimizer_runs", "optimizer-runs"], default_profile)
-            if optimizer_runs is None:
-                # Default to 200
-                result.optimizer_runs = 200
-            else:
-                result.optimizer_runs = optimizer_runs
-            evm_version = lookup_by_keys(["evm_version", "evm-version"], default_profile)
-            if evm_version is None:
-                result.evm_version = evm_version
-            else:
-                # Default to london
-                result.evm_version = "london"
-            if "src" in default_profile:
-                result.src_path = default_profile["src"]
-            if "test" in default_profile:
-                result.tests_path = default_profile["test"]
-            if "libs" in default_profile:
-                result.libs_path = default_profile["libs"]
-            if "script" in default_profile:
-                result.scripts_path = default_profile["script"]
+        result.remappings = json_config["remappings"]
+
+        def lookup_by_keys(keys: List[str], dictionary: Dict[str, T]) -> Optional[T]:
+            for key in keys:
+                if key in dictionary:
+                    return dictionary[key]
+            return None
+
+        # Foundry supports snake and kebab case.
+        result.solc_version = lookup_by_keys(["solc", "solc_version", "solc-version"], json_config)
+        via_ir = lookup_by_keys(["via_ir", "via-ir"], json_config)
+        if via_ir:
+            result.via_ir = via_ir
+        result.allow_paths = lookup_by_keys(["allow_paths", "allow-paths"], json_config)
+
+        if "offline" in json_config:
+            result.offline = json_config["offline"]
+        if "optimizer" in json_config:
+            result.optimizer = json_config["optimizer"]
+        else:
+            # Default to true
+            result.optimizer = True
```

Comment:
If you're looking to provide a default value, you can use [`dict.get()`](https://docs.python.org/3/library/stdtypes.html#dict.get), which results in more concise code. You could also implement a similar behavior for `lookup_by_keys` by adding a new `default=None` argument and returning it in the end instead of the hardcoded `None`.

```suggestion
        result.optimizer = json_config.get("optimizer", True)
```